packages/relay/src/lib/eth.ts

```diff
@@ -1146,28 +1146,59 @@ export class EthImpl implements Eth {
         throw e;
       }
     } else if (fromBlock || toBlock) {
-      const filters = [];
-      let order;
+      let fromBlockTimestamp;
+      let toBlockTimestamp;
+      let fromBlockNum = 0;
+      let toBlockNum;
+
       if (toBlock) {
-        // @ts-ignore
-        filters.push(`lte:${parseInt(toBlock)}`);
-        order = constants.ORDER.DESC;
+        try {
+          const blockResponse = await this.getHistoricalBlockResponse(toBlock, true, requestId);
+          toBlockTimestamp = blockResponse.timestamp.to;
+          toBlockNum = parseInt(blockResponse.number);
+        } catch (e) {
+          // Check if block error is RESOURCE_NOT_FOUND, the most likely scenario this will happen if a block number > height is passed
+          // In this case eth_getLogs() returns logs up to the latest block.
+          // @ts-ignore
+          if (e.statusCode != 404) {
+            throw e;
+          }
+        }
       }
       if (fromBlock) {
-        // @ts-ignore
-        filters.push(`gte:${parseInt(fromBlock)}`);
-        order = constants.ORDER.ASC;
+        try {
+          const blockResponse = await this.getHistoricalBlockResponse(fromBlock, undefined, requestId);
+          fromBlockTimestamp = blockResponse.timestamp.from;
+          fromBlockNum = parseInt(blockResponse.number);
+        } catch (e) {
+          // Check if block error is RESOURCE_NOT_FOUND, the most likely scenario this will happen if a block number > height is passed
+          // In this case eth_getLogs() returns an empty array.
+          // @ts-ignore
+          if (e.statusCode != 404) {
```

Comment:
if 404 use `predefined.RESOURCE_NOT_FOUND`

Reply:
Same as the previous comment

Reply:
Okay.
Maybe add a comment